assignments/closure.js

```diff
@@ -1,20 +1,53 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
 
+function movies(title) {
+  const movieName = title;
+  console.log(`My favorite movie is ${movieName}`)
+  
+  function actor() {
+    const favActor = "Denzel Washington";
+    console.log(`My favorite actor is ${favActor} and my favorite movie is ${movieName}`);
+
+    function director() {
+      const favDir = "Frank Capra";
+      console.log(`${favDir} is my favorite movie director, ${favActor} is my favorite movie actor, and ${movieName} is my favorite movie. `)
+    }
+    director();
+  }
+  actor();
+}
+
+movies("Braveheart");
 
 // ==== Challenge 2: Create a counter function ====
 const counter = () => {
-  // Return a function that when invoked increments and returns a counter variable.
+  let count = 0;
+  return () => (++count)
+  
 };
-// Example usage: const newCounter = counter();
-// newCounter(); // 1
-// newCounter(); // 2
+
+const newCounter = counter();
+console.log(newCounter());
+console.log(newCounter());
+console.log(newCounter());
+console.log(newCounter());
+
 
 /* STRETCH PROBLEM, Do not attempt until you have completed all previous tasks for today's project files */
 
 // ==== Challenge 3: Create a counter function with an object that can increment and decrement ====
 const counterFactory = () => {
+  let count = 0;
+  return () => (++count)
```

Comment:
Your first return only will be returned, this way you are to able to decrement
What you can do is return a object with increment and decrement functions 

```
const counterFactory = () => {
  let counter = 0;
  return {
   increment: () => ++counter,
   decremnt: () => --counter
  }
}
```